### PR TITLE
Basic tactic failing condition tests

### PR DESCRIPTION
**Pull request by Ratoone**
As the tactic says. The state machine reset tests are not implemented though, as they should be performed directly on the state machine

### CODE REVIEW

**Comment by Ratoone** [comment]
This is part of #880

**Review by timovdk — APPROVED** [review]
Looks good

File: test/StpTests/TacticTests.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    robot_proto.set_id(1);
+    robot_proto.mutable_pos()->set_x(0.1);
+    std::unordered_map<uint8_t, proto::RobotFeedback> updateMap;
+    auto robot = rtt::world_new::robot::Robot(updateMap, robot_proto);
+    info.setRobot(rtt::world_new::view::RobotView(&robot));
+
+    EXPECT_CALL(tactic, isEndTacticMock()).WillOnce(testing::Return(true));
+    EXPECT_CALL(tactic, isTacticFailingMock(testing::_)).WillOnce(testing::Return(true));
+    auto result = tactic.update(info);
+    ASSERT_EQ(rtt::ai::stp::Status::Failure, result);
```
Comment (RobotJente) [inline test/StpTests/TacticTests.cpp:79]: It is not clear to me what this part is testing. Can you put some comments to explain what is going on here?

File: test/StpTests/TacticTests.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    robot_proto.set_id(1);
+    robot_proto.mutable_pos()->set_x(0.1);
+    std::unordered_map<uint8_t, proto::RobotFeedback> updateMap;
+    auto robot = rtt::world_new::robot::Robot(updateMap, robot_proto);
+    info.setRobot(rtt::world_new::view::RobotView(&robot));
+
+    EXPECT_CALL(tactic, isEndTacticMock()).WillOnce(testing::Return(true));
+    EXPECT_CALL(tactic, isTacticFailingMock(testing::_)).WillOnce(testing::Return(false));
+    auto result = tactic.update(info);
+    ASSERT_EQ(rtt::ai::stp::Status::Running, result);
```
Comment (RobotJente) [inline test/StpTests/TacticTests.cpp:99]: Same here, can you put a brief summary of what you test and how it's tested here?

**Comment by Ratoone** [comment]
Added extra comments to explain what's happening & merged

**Pull request merged**